contracts/pcv/utils/NamedStaticPCVDepositWrapper.sol

```diff
@@ -0,0 +1,202 @@
+
+        uint256 oldBalance = balance;
+        uint256 oldFeiReportBalance = feiReportBalance;
+
+        uint256 lastIndex = pcvDeposits.length - 1;
+
+        if (lastIndex != index) {
+            DepositInfo storage lastvalue = pcvDeposits[lastIndex];
+
+            // Move the last value to the index where the value to delete is
+            pcvDeposits[index] = lastvalue;
+        }
+        // Delete the slot where the moved value was stored
+        pcvDeposits.pop();
+
+        balance -= depositBalance;
+        feiReportBalance -= feiDepositBalance;
+
+        emit BalanceUpdate(oldBalance, balance, oldFeiReportBalance, feiReportBalance);
+        emit DepositRemoved(index);
+    }
+
+    /// @notice helper method to add a PCV deposit
+    function _addDeposit(DepositInfo memory newPCVDeposit) internal {
+        require(newPCVDeposit.feiAmount > 0 || newPCVDeposit.usdAmount > 0, "NamedStaticPCVDepositWrapper: must supply either fei or usd amount");
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance += newPCVDeposit.usdAmount;
+        feiReportBalance += newPCVDeposit.feiAmount;
+        pcvDeposits.push(newPCVDeposit);
+
+        emit DepositAdded(pcvDeposits.length - 1, newPCVDeposit.depositName);
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    /// @notice helper method to edit a PCV deposit
+    function _editDeposit(
+        uint256 index,
+        string calldata depositName,
+        uint256 usdAmount,
+        uint256 feiAmount,
+        uint256 underlyingTokenAmount,
+        address underlyingToken
+    ) internal {
+        DepositInfo storage updatePCVDeposit = pcvDeposits[index];
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance = balance - updatePCVDeposit.usdAmount + usdAmount;
+        feiReportBalance = feiReportBalance - updatePCVDeposit.feiAmount + feiAmount;
+
+        updatePCVDeposit.usdAmount = usdAmount + updatePCVDeposit.usdAmount;
+        updatePCVDeposit.depositName = depositName;
+        updatePCVDeposit.underlyingTokenAmount = underlyingTokenAmount;
+        updatePCVDeposit.underlyingToken = underlyingToken;
+
+        emit DepositEdited(index, depositName);
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    // ----------- Governor only state changing api -----------
+
+    /// @notice function to add a deposit
+    function addDeposit(
+        DepositInfo calldata newPCVDeposit
+    ) external onlyGovernorOrAdmin {
+        _addDeposit(newPCVDeposit);
+    }
+
+    /// @notice function to bulk add deposits
+    function bulkAddDeposits(
+        DepositInfo[] calldata newPCVDeposits
+    ) external onlyGovernorOrAdmin {
+        for (uint256 i = 0; i < newPCVDeposits.length; i++) {
+            _addDeposit(newPCVDeposits[i]);
+        }
+    }
```

Comment:
Add a bulk remove method and we good I think

Reply:
The reason this doesn't work is that if you pass an array of indexes to remove, by the time you remove the first one, all of the other ones will be incorrect. The only bulk remove method that works would be one that deletes all items.

Reply:
++

this isn't simple though

Reply:
It could work if the array of indexes was properly ordered from greatest indexes to least, but if unordered or not sorted, it does not work.